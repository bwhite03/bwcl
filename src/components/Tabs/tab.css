.tab-list {
  border-bottom: 1px solid #ccc;
  padding-left: 0;
}

.tab-list-item {
  display: inline-block;
  list-style: none;
  margin-bottom: -1px;
  cursor: pointer;
  padding: 5px;
}

.tab-list-active {
  background-color: #a82cc2;
  border: solid #ccc;
  border-width: 1px 1px 0 1px;
  color: #fff;
}
