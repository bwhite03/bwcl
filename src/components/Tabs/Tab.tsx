import React from "react";
import { TableProps } from "../Grid/DataGrid";
import "./tab.css";

export interface TabProps {
  className: string;
  onClick: (e: string) => void;
  label: string;
  activeTab: string;
  disabled: boolean;
  id: string;
}

const Tab = (props: TabProps) => {
  let className = "tab-list-item";
  if (props.label === props.activeTab) {
    className += " tab-list-active";
  }

  return (
    <li
      onClick={() => props.onClick(props.label)}
      className={className}
      id={props.id}
    >
      {props.label}
    </li>
  );
};

export default Tab;
