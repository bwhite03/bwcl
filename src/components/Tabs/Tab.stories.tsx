import React from "react";
import { Story, Meta } from "@storybook/react";

import Tab, { TabProps } from "./Tab";

export default {
  title: "Tab",
  component: Tab,
} as Meta;

const Template: Story<TabProps> = (args) => <Tab {...args} />;

export const Tab1 = Template.bind({});

Tab1.args = {
  label: "Tab1",
};

export const Tab2 = Template.bind({});
Tab2.args = {
  label: "Tab2",
};
