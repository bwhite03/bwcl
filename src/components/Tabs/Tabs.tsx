import React, { PropsWithChildren, useState } from "react";
import Tab from "./Tab";
import "./tab.css";

export interface TabsProps {
  children: JSX.Element[];
}

const Tabs = (props: TabsProps) => {
  const [activeTab, setActiveTab] = useState<string>("");

  const handeTabClick = (tab: string) => {
    setActiveTab(tab);
  };

  return (
    <div className="tabs">
      <ol className="tab-list">
        {props.children &&
          props.children.map((child, index) => {
            if (child) {
              return (
                <Tab
                  activeTab={activeTab}
                  key={`tabs-li-${index}`}
                  label={child.props.label}
                  disabled={child.props.disabled}
                  onClick={handeTabClick}
                  id={child.props.id}
                  className={child.props.className}
                />
              );
            }
          })}
      </ol>
      <div className="tab-content">
        {props.children &&
          props.children.map((child, idx) => {
            if (child.props.label !== activeTab) return undefined;
            return child.props.children;
          })}
      </div>
    </div>
  );
};

export default Tabs;
