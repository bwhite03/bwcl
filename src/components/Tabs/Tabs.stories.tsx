import React from "react";
import { Story, Meta } from "@storybook/react";

import Tabs, { TabsProps } from "./Tabs";
import { Tab1, Tab2 } from "./Tab.stories";

export default {
  title: "Tabs",
  component: Tabs,
} as Meta;

const Template: Story<TabsProps> = (args) => <Tabs {...args} />;

export const TabsExample = Template.bind({});
TabsExample.args = {
  // @ts-ignore
  children: [<Tab1 {...Tab1.args} />, <Tab2 {...Tab2.args} />],
};

export const Tab1Example = Template.bind({});
Tab1Example.args = {
  children: [
    // @ts-ignore
    <div label="tab1">
      <h2>Tab 1</h2>
    </div>,
    // @ts-ignore
    <div label="tab2">
      <h2>Tab 2</h2>
    </div>,
  ],
};
