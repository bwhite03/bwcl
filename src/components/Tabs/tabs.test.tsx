// @ts-nocheck
import React from "react";

import { render, fireEvent } from "@testing-library/react";
import Tabs from "./Tabs";

describe("Tabs", () => {
  test("should render correctly", () => {
    const testFn = jest.fn();
    const { container } = render(
      <Tabs>
        <div label="tab1" name="tab1">
          <h2>Tab1</h2>
        </div>
        <div label="tab2" name="tab2">
          <h2>Tab2</h2>
        </div>
      </Tabs>
    );
    expect(container).toMatchSnapshot();
  });

  test("should handle click", () => {
    const testFn = jest.fn();
    const { container, getByRole } = render(
      <Tabs>
        <div label="tab1" name="tab1">
          <h2>Tab1</h2>
        </div>
        <div label="tab2" name="tab2">
          <h2>Tab2</h2>
        </div>
      </Tabs>
    );

    const tab = container.querySelector("div > div:nth-child(1)");
    // @ts-ignore
    fireEvent.click(tab);
  });
});
