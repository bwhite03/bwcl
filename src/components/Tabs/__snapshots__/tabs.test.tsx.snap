// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tabs should handle click 1`] = `
<div>
  <div
    class="tabs"
  >
    <ol
      class="tab-list"
    >
      <li
        class="tab-list-item"
      >
        tab1
      </li>
      <li
        class="tab-list-item"
      >
        tab2
      </li>
    </ol>
    <div
      class="tab-content"
    />
  </div>
</div>
`;

exports[`Tabs should render correctly 1`] = `
<div>
  <div
    class="tabs"
  >
    <ol
      class="tab-list"
    >
      <li
        class="tab-list-item"
      >
        tab1
      </li>
      <li
        class="tab-list-item"
      >
        tab2
      </li>
    </ol>
    <div
      class="tab-content"
    />
  </div>
</div>
`;
